ethcore/src/header.rs

```diff
@@ -186,11 +186,12 @@ impl Header {
 	/// Get the hash of the header excluding the seal
 	pub fn bare_hash(&self) -> H256 {
 		let mut hash = self.bare_hash.borrow_mut();
-		match &mut *hash {
+		match hash.deref_mut() {
```

Comment:
> so it is strictly more flexible

rather going against the "everything explicit" philosophy that much of the rest of the language seems to have...